### PR TITLE
Add a custom sliding sync proxy setting in developer options

### PR DESCRIPTION
**Pull request by jmartinesp**
## Content

Added a new `customSlidingSyncProxy` preference in `AppPreferencesStore` that will be loaded when restoring the session so it can be used to test a different proxy.

## Motivation and context

Implements the Android part of https://github.com/element-hq/element-meta/issues/2468

## Tests

- Go to developer options, Rust SDK section.
- Select `custom sliding sync proxy`.
- Set a custom value.
- Kill and restart the app, the new proxy should be used.
- Undo what you did by deleting the custom proxy text and restarting the app.
- The default proxy should be used.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 14

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request title will be used in the release note, it clearly define what will change for the user
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: libraries/preferences/impl/src/main/kotlin/io/element/android/libraries/preferences/impl/store/DefaultSessionPreferencesStore.kt [hunk 1]
```diff
@@ -49,7 +49,7 @@ class DefaultSessionPreferencesStore(
     private val renderReadReceiptsKey = booleanPreferencesKey("renderReadReceipts")
     private val sendTypingNotificationsKey = booleanPreferencesKey("sendTypingNotifications")
     private val renderTypingNotificationsKey = booleanPreferencesKey("renderTypingNotifications")
-    private val skipSessionVerification = booleanPreferencesKey("skipSessionVerification")
+    private val skipSessionVerificationKey = booleanPreferencesKey("skipSessionVerification")
```
Comment (bmarty) [inline libraries/preferences/impl/src/main/kotlin/io/element/android/libraries/preferences/impl/store/DefaultSessionPreferencesStore.kt:52]: 👍

**Review by bmarty — APPROVED** [review]
LGTM, thanks a lot!

**Comment by jmartinesp** [comment]
Apparently this PR doesn't match what we want to do since we want to change a part of the sliding sync request URL that can't be overridden with this new option. I'm wondering if we should still merge it in case we'll find some use for it in the future 🤔 .

File: libraries/preferences/impl/src/main/kotlin/io/element/android/libraries/preferences/impl/store/DefaultSessionPreferencesStore.kt (hunk 1 above)
Comment (bmarty) [reply libraries/preferences/impl/src/main/kotlin/io/element/android/libraries/preferences/impl/store/DefaultSessionPreferencesStore.kt:52]: (but ideally this renaming should be done in a separate commit: for next time!)

**Comment by bmarty** [comment]
> we want to change a part of the sliding sync request URL

I am wondering how a user could customize this at runtime.

**Comment by jmartinesp** [comment]
> I am wondering how a user could customize this at runtime.

That's the thing, we can't. That's hardcoded in Ruma, and that's why this PR didn't fit what we needed. It's probably better if I just close it.

**Pull request closed without merging**